nexus/src/lib.rs

```diff
@@ -14,19 +14,18 @@
 
 pub mod app; // Public for documentation examples
 mod cidata;
-pub mod config; // Public for testing
-pub mod context; // Public for documentation examples
+mod config;
+mod context; // Public for documentation examples
 pub mod external_api; // Public for testing
-pub mod internal_api; // Public for testing
+mod internal_api;
 mod populate;
 mod saga_interface;
-pub mod updates; // public for testing
+mod updates; // public for testing
 
 pub use app::test_interfaces::TestInterfaces;
 pub use app::Nexus;
```

Comment:
I normally wouldn't rely on `pub(crate)` as much as I do in this PR, but this exposure (as well as `pub mod app`), which is used by the integration tests, makes it hard to determine "where things would be exposed out of the crate" at a module level.